x/feedist/handler.go

```diff
@@ -131,11 +141,50 @@ func (h *updateRevenueHandler) validate(ctx weave.Context, db weave.KVStore, tx
 // distribute split the funds stored under the revenue address and distribute
 // them according to recipients proportions. When successful, revenue account
 // has no funds left after this call.
-func distribute(ctx weave.Context, db weave.KVStore, rev *Revenue) error {
-	var chunks int32
-	for _, r := range rev.Recipients {
-		chunks += r.Weight
+//
+// It might be that not all funds can be distributed equally. Because of that a
+// small leftover can remain on the revenue account after this operation.
+func distribute(db weave.KVStore, ctrl CashController, source weave.Address, recipients []*Recipient) error {
+	var chunks int64
+	for _, r := range recipients {
+		chunks += int64(r.Weight)
 	}
 
-	panic("todo")
+	balance, err := ctrl.Balance(db, source)
+	if err != nil {
+		return errors.Wrap(err, "cannot acquire revenue account balance")
+	}
+
+	// TODO normalize balance. There is no functionality that allows to
+	// normalize x.Coins right now (14 Feb 2019).
+
+	// For each currency, distribute the coins equally to the weight of
+	// each recipient. This can leave small amount of coins on the original
+	// account.
+	for _, c := range balance {
+		// Ignore those coins that have a negative value. This
+		// functionality is supposed to be distributing value from
+		// revenue account, not collect it. Otherwise this could be
+		// used to charge the recipients instead of paying them.
+		if !c.IsPositive() {
+			continue
+		}
+
+		for _, r := range recipients {
+			amount := x.Coin{
+				Whole:      (c.Whole / chunks) * int64(r.Weight),
```

Comment:
This is not good distribution... you need to carry.

Let's say I have 4 rcpts with weight 1 and want to split out 5 IOV...
Each should get 1.25 (`{Whole: 1, Fractional: 250000000}`)
This code would give each exactly 1.

"Just" got to carry the remainder :)

Reply:
Good point. The information how many fractional makes a whole is stored in the [`x/currency.TickerInfo`](https://github.com/iov-one/weave/blob/master/x/currency/codec.proto#L7), right? I have to get this information from there and apply.

Reply:
This discussion is continued in https://github.com/iov-one/weave/pull/316#discussion_r256932551